### ISSUE TITLE
"Cannot find control with unspecified name attribute" plz add the name of the control

### ISSUE THREAD

**Issue by phil123456** [angular/angular#19153]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

concerning error "Cannot find control with unspecified name attribute"

can someone add the actual name angular is looking for ?
that would be awesome especialy when you have like 20 inputs in a form
and try to find which one is actually missing in the template

thanks

**Comment by djleonskennedy** [comment]
@phil123456 guess it isn't feature request its a bug

**Comment by kara** [comment]
Yeah, we should improve this error message. The problem is that we throw this particular one when the control does *not* have a name, so we can't provide one.

**Comment by timcroydon** [comment]
For anyone stumbling across this error, I found it was caused in my case by using `[formControl]='myControlName'` rather than `formControlName='myControlName'` after I rearranged my form controls into a form group in the component code and forgot to change the directive.

I agree the error message could be more useful, as I spent far too long double-checking control spellings and overlooked the actual directive name.

**Comment by amitsbaghel** [comment]
formControlName='myControlName' worked for me. Thanx

**Comment by Anirudh16** [comment]
got the error when using Formgroup

<form [formGroup]="myvehicle"></form>

may I know where is the mistake

**Comment by Yogi2001** [comment]
I ran into this Error in various ways
1) no getter for the FormControl `  get solarpannels(): AbstractControl {
      return this.form.get('solarpannels');
  }
`
2) typo in in either 
  2a) name
  2b) getter
  2c) creation of the form group

3) missing in form-group element when creating the form

**Comment by mpeguero** [comment]
This is the way i Solved my problem, It works for me:
TS:
get OptionsControls() {
    return (this.form.get("options") as FormGroup).controls;
  }

  get options(): FormArray {
    return (this.form.get("options") as FormArray);
  }
 Template:
  <div formArrayName="options" >
`  <div 
            fxFlex
            fxLayout="row"
            *ngFor="let item of OptionsControls; let i = index"
            style="padding: 0 10px 0 10px;"
          >
            <div fxLayout="row" fxFlex fxLayoutGap="2%" >
              <mat-form-field fxFlex>
                <mat-label>Option {{ i + 1 }}</mat-label>
                <input
                  type="text"
                  matInput
                  name="opcion"
                  required
                  [formControl]="item"
                  placeholder="Enter a option"
                />
              </mat-form-field>
                  </div>
          </div>
</div>`

**Comment by thw0rted** [comment]
Can someone from the team update this with a status?  I've just been bitten by this and the problem was simply that I misspelled the name of the property passed, e.g. `[formControl]="myFormControol"`.  You're not looking for a "control with unspecified name attribute", you're being passed an undefined control.

If the way the directive is invoked makes it impossible to tell the difference between `<input [formControlName]="undefined">` and `<input [formControl]="undefined">`, then the error message should reflect that.  (I would argue that *both* of those should actually behave identically to an input with *no* reactive form directive, but maybe that's another issue?)

**Comment by sbettwieser** [comment]
Someone please make this error clearer. I've got so many form controls in my project and the stack trace points to a line which doesn't have one, so I'm not sure which one is triggering it.

**Comment by joypicar** [comment]
this works for me:

in html:
[formControl]="$any(control)"

in ts:
control: AbstractControl<any> = new FormControl();

**Comment by mwaibel-go** [comment]
I know too little about the inner workings of Angular to see if this is a general solution. Maybe others can add if this is helpful to them:

The second-to-last frame in the stack trace is this function:

https://github.com/angular/angular/blob/f0965c7acd2fc2a4a4c18e5a47f3447c4fc7c668/packages/forms/src/directives/shared.ts#L66-L72

When you investigate `dir`, the directive to which Angular _tries_ to attach the FormControl, you may find clues about which HTML element is being rendered. I found the native HTMLInputElement in `dir.valueAccessor._elementRef.nativeElement`.

If this is a general solution, maybe the ID or `outerHTML` of that element could be used in the error message.
